### DISCUSSION TITLE
Getting a error to upgrade the @zowe/imperative version to 6.24.1Te

### DISCUSSION THREAD

**Discussion by ghost** [zowe/zowe-cli#1560] [Q&A]
Hi Team, 
I have updated my application with below package.json and did a build through jenkins and it was successful. 
Then i installed "NPM INSTALL " in my nexus repository followed by issued "npm audit fix --force". and its popping up an error that @zowe/imperative version 6.24.1 not found. Need your help fixing this. 

My Package.json has the below dependencies :

"dependencies": {
    "@zowe/cli": "^7.6.2",
	"@zowe/imperative": "^5.5.4",
    "@zowe/zos-jobs-for-zowe-sdk": "^7.6.2",
    "@zowe/zos-tso-for-zowe-sdk": "^7.6.2",
    "@zowe/zos-uss-for-zowe-sdk": "^7.6.2",
	"@zowe/zos-files-for-zowe-sdk": "^7.7.0",
    "body-parser": "^1.20.1",
    "cookie-parser": "~1.4.4",
    "cors": "^2.8.5",
    "debug": "~2.6.9",
    "dotenv": "^16.0.3",
    "express": "~4.16.1",
    "express-zip": "^3.0.0",
    "helmet": "^5.1.1",
    "jszip": "^3.10.1",
    "morgan": "~1.10.0",
    "tingodb": "^0.6.1"
  },
  "devDependencies": {
    "nodemon": "^2.0.20"

error msg : 
$ npm audit fix --force @zeng/zoops-app-server@1.0.51
npm WARN using --force Recommended protections disabled.
npm WARN audit No fix available for express@*
npm WARN audit No fix available for debug@*
npm WARN audit Updating nodemon to 1.9.0, which is a SemVer major change.
npm WARN audit Updating @zowe/imperative to 6.24.1, which is a SemVer major change.
npm WARN audit Updating cookie-parser to 1.0.0, which is a SemVer major change.
npm WARN audit No fix available for express-zip@*
npm WARN audit No fix available for body-parser@*
npm WARN audit Updating morgan to 1.2.2, which is a SemVer major change.
npm WARN audit Updating cors to 2.8.1, which is a SemVer major change.
npm WARN audit Updating jszip to 3.9.1, which is a SemVer major change.
npm WARN audit Updating @zowe/zos-uss-for-zowe-sdk to 6.24.1, which is a SemVer major change.
npm WARN audit Updating @zowe/zos-tso-for-zowe-sdk to 6.24.1, which is a SemVer major change.
npm WARN audit Updating @zowe/zos-jobs-for-zowe-sdk to 6.24.1, which is a SemVer major change.
npm WARN audit No fix available for @zowe/cli@*
npm WARN audit Updating @zowe/zos-files-for-zowe-sdk to 6.24.1, which is a SemVer major change.
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @zowe/imperative@6.24.1.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\F8512PV\AppData\Local\npm-cache\_logs\2022-11-07T10_03_11_737Z-debug-0.log

[2022-11-07T10_03_11_737Z-debug-0.log](https://github.com/zowe/zowe-cli/files/9950319/2022-11-07T10_03_11_737Z-debug-0.log)

**Comment by t1m0thyj** [comment]
Currently the latest version of the `@zowe/imperative` package is 5.7.0 (see [public npm](https://www.npmjs.com/package/@zowe/imperative)).

I'm not sure why the npm audit would think there is a version 6.24.1 of Imperative. Could you run the command `npm view @zowe/imperative` against your Nexus repository to see what it thinks the latest version of Imperative is?

**Reply by ghost** [reply]
Hi Timothy, Thanks for looking into it. 

below is the command output. 

 npm view @zowe/imperative

@zowe/imperative@5.7.0 | EPL-2.0 | deps: 36 | versions: 191
framework for building configurable CLIs
https://github.com/zowe/imperative#readme

keywords: CLI, framework, zowe

dist
.tarball: https://nexus-appdev.1dc.com/repository/npm-group/@zowe/imperative/-/imperative-5.7.0.tgz
.shasum: 161e9acbb098c4905832228b72055160fd443d4f
.integrity: sha512-r04AT8hquSqBwmUhTYRRcUst+6snnJilql+dFg+VlVTcijhJHIfy7H1OJNimMyHT5XTHB2uaQU0c7HLNqbRR6g==
.unpackedSize: 3.9 MB

dependencies:
@types/yargs: 13.0.4
@zowe/perf-timing: 1.0.7
chalk: 2.4.2
cli-table3: 0.6.2
comment-json: 4.1.1
dataobject-parser: 1.2.1
deepmerge: 4.2.2
diff2html: 3.4.20-usewontache.1.60e7a2e
diff: 5.1.0
fast-glob: 3.2.7
fastest-levenshtein: 1.0.12
find-up: 4.1.0
fs-extra: 8.1.0
glob: 7.2.3
jest-diff: 27.0.6
js-yaml: 4.1.0
jsonfile: 4.0.0
jsonschema: 1.1.1
lodash-deep: 2.0.0
lodash: 4.17.21
log4js: 6.4.6
markdown-it: 12.3.2
mustache: 2.3.0
npm-package-arg: 9.1.0
(...and 12 more.)

maintainers:
- joe_winchester <winchest@uk.ibm.com>
- zowerobot <zowe.robot@gmail.com>
- markackertbroadcom <Mark.Ackert@broadcom.com>

dist-tags:
latest: 5.7.0
next: 5.7.0
zowe-v1-lts: 4.18.9
zowe-v2-lts: 5.7.0

published 5 days ago by zowerobot <zowe.robot@gmail.com>
